lib/ModelMixins/GeojsonMixin.ts

```diff
@@ -150,6 +159,51 @@ export default function GeoJsonMixin<
         });
     }
 
+    @action
+    private addPerPropertyStyleToGeoJson(json: JsonObject | GeoJsonObject) {
+      const geojson = json as GeoJsonObject;
+      if (geojson.type === "Feature") {
+        const featureProperties = (geojson as Feature).properties;
+        if (featureProperties === null) {
+          return;
+        }
+        const featurePropertiesEntires = Object.entries(featureProperties);
+
+        const matchedStyles = this.perPropertyStyles.filter(style => {
+          const stylePropertiesEntries = Object.entries(
+            style.properties as any
+          );
+
+          // For every key-value pair in the style, is there an identical one in the feature's properties?
+          return stylePropertiesEntries.every(
+            ([styleKey, styleValue]) =>
+              featurePropertiesEntires.find(([featKey, featValue]) => {
+                if (typeof styleValue === "string" && !style.caseSensitive) {
+                  featKey === styleKey &&
+                    (featValue as string).toLowerCase() ===
+                      (styleValue as string).toLowerCase();
+                }
+                return featKey === styleKey && featValue === styleValue;
+              }) !== undefined
+          );
+        });
+
+        if (matchedStyles !== undefined) {
+          for (let matched of matchedStyles) {
+            for (let trait of Object.keys(matched.style.traits)) {
+              // @ts-ignore - TS can't tell that `trait` is of the correct index type for style
+              featureProperties[trait] = matched.style[trait];
```

Comment:
This will overwrite values with `undefined`

Whereas this would keep existing values:

```suggestion
              const styleValue = matched.style[trait];
              if (isDefined(styleValue)) featureProperties[trait] = styleValue;
```

Reply:
Fixed! But the `@ts-ignore` is still necessary, typescript still complains